sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,502 @@
+        return psol, linsolve_dict(auxeq.all_coeffs(), psyms, x), True
+    else:
+        # m == 0 . Check if 1 (x**0) is a solution to the auxiliary equation
+        return S(1), auxeq, auxeq == 0
+
+
+def solve_riccati(fx, x, b0, b1, b2):
+    # Step 1 : Convert to Normal Form
+    a = -b0*b2 + b1**2/4 - b1.diff(x)/2 + 3*b2.diff(x)**2/(4*b2**2) + b1*b2.diff(x)/(2*b2) - b2.diff(x, 2)/(2*b2)
+    a_t = a.together()
+    num, den = [Poly(e, x, extension=True) for e in a_t.as_numer_denom()]
+    num, den = num.cancel(den, include=True)
+
+    # Step 2
+    presol = []
+
+    # Step 3 : a(x) is 0
+    if num == 0:
+        presol.append(1/(x + Dummy('C1')))
+
+    # Step 4 : a(x) is a non-zero constant
+    elif x not in num.free_symbols.union(den.free_symbols):
+        presol.extend([sqrt(a), -sqrt(a)])
+
+    # Step 5 : Find poles and valuation at infinity
+    poles = find_poles(num, den, x)
+    inf_mul = poles.get(oo, 0)
+    poles, muls = list(poles.keys()), list(poles.values())
+    val_inf = val_at_inf(num, den, x)
+
+    if len(poles):
+        # Check necessary conditions (outlined in the module docstring)
+        if check_necessary_conds(val_inf, muls):
+            raise ValueError("Rational Solution doesn't exist")
+
+        # Step 6
+        # Construct c-vectors for each singular point
+        c = construct_c(num, den, x, poles, muls)
+
+        # Construct d vectors for each singular point
+        d = construct_d(num, den, x, val_inf, inf_mul)
+
+        # Step 7 : Iterate over all possible combinations and return solutions
+        for it in range(2**(len(poles) + 1)):
+            # For each possible combination, generate an array of 0's and 1's
+            # where 0 means pick 1st choice and 1 means pick the second choice.
+            choice = [int(x) for x in bin(it)[2:].zfill(len(poles) + 1)]
+
+            # Step 8 and 9 : Compute m and ybar
+            m, ybar, numy, deny, exists = compute_degree(x, poles, choice, c, d, -val_inf//2)
+
+            # Step 10 : Check if a valid solution exists. If yes, also check
+            # if m is a non-negative integer
+            if exists and S(m).is_nonnegative == True and S(m).is_integer == True:
+
+                # Step 11 : Find polynomial solutions of degree m for the auxiliary equation
+                psol, coeffs, exists = solve_aux_eq(num, den, numy, deny, x, m)
+
+                # Step 12 : If valid polynomial solution exists, append solution.
+                if exists:
+                    # m == 0 case
+                    if psol == 1 and coeffs == 0:
+                        # p(x) = 1, so p'(x)/p(x) term need not be added
+                        presol.append(ybar)
+                    # m is a positive integer and there are valid coefficients
+                    elif len(coeffs):
+                        # Substitute the valid coefficients to get p(x)
+                        psol = psol.subs(coeffs)
```

Comment:
In some cases `coeffs` here is a large dict. In those cases it is more efficient to use `xreplace` rather than `subs`. In general the reason for preferring subs is because it handles bound vs free symbols but I don't think that is an issue here. Some slow examples can be made much faster by using `xreplace` here.